From: srinivas pulukuru <srinivas.pulukuru@ti.com>
Date: Tue, 3 May 2011 16:42:19 +0000 (-0500)
Subject: OMAP4 DSS HDMI: Fix HDMI connect - disconnect path upon resume
X-Git-Url: http://omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=c298f7db0489188ae1d6dfb2a7861a1617664f94

OMAP4 DSS HDMI: Fix HDMI connect - disconnect path upon resume

1. When the HDMI resumes, we see HDMI_FIRST_HPD after disconnect.
This should be ignored and interrupts cleared before returning.
2. Phy connect leaves hdmi_power in HDMI_POWER_FULL stand
disconnect leaves the power state in HDMI_POWER_MIN. when power state
is HDMI_POWER_MIN, first_hpd should not be processed.

Change-Id: Ibf216ff0ab5a33e2182898fd1721bafdddae14ed
Signed-off-by: srinivas pulukuru <srinivas.pulukuru@ti.com>
---

diff --git a/drivers/video/omap2/dss/hdmi.c b/drivers/video/omap2/dss/hdmi.c
index 594bc47..c920da7 100644
--- a/drivers/video/omap2/dss/hdmi.c
+++ b/drivers/video/omap2/dss/hdmi.c
@@ -1527,7 +1527,7 @@ done:
 		 */
 		DSSINFO("Enabling display - HDMI_FIRST_HPD\n");
 
-		if (!hdmi_connected) {
+		if (!hdmi_connected || (hdmi_power == HDMI_POWER_MIN)) {
 			DSSINFO("irqstatus=0x%08x ignoring FIRST_HPD when "
 				"hdmi_connected = %d, hdmi_power = %d\n",
 				r, hdmi_connected, hdmi_power);